include/nlohmann/detail/conversions/to_json.hpp

```diff
@@ -164,6 +167,9 @@ struct external_constructor<value_t::array>
         for (const bool x : arr)
         {
             j.m_value.array->push_back(x);
+#if JSON_DIAGNOSTICS
+            j.m_value.array->back().m_parent = &j;
```

Comment:
`j.set_parent(...`

Reply:
Fixed in 0d1fb383b7f4572a154d7a26a1318e2c617b4de8.